src/ccache.c

```diff
@@ -3255,7 +3262,7 @@ cc_process_args(struct args *args, struct args **preprocessor_args,
 		goto out;
 	}
 
-	if (!found_c_opt && !found_S_opt) {
+	if (!found_c_opt && !found_dc_opt && !found_S_opt) {
 		if (output_is_precompiled_header) {
```

Comment:
I'm a bit unsure about this. I don't think NVCC does precompiled headers at all, so I don't know what the conditions are for `output_is_precompiled_header` to become true.